src/Model/Behavior/SlugBehavior.php

```diff
@@ -194,11 +197,12 @@ public function beforeSave(Event $event, Entity $entity, ArrayObject $options)
         $fields = (array)$config['displayField'];
         $parts = [];
         foreach ($fields as $field) {
-            if (!isset($entity->{$field}) && !$entity->isNew()) {
+            $value = Hash::get($entity, $field);
+
+            if ($value === null && !$entity->isNew()) {
```

Comment:
I would prefer not to add version check. We can make a 3.0 compatible release prior to merging this and then bump plugin version. In future in case a bug is reported we can make bugfix releases for the 3.0 compatible version. It unlikely we are going to get too many issues so maintaining older version too shouldn't be much of a chore.